quint/src/quintParserFrontend.ts

```diff
@@ -128,6 +129,101 @@ export function parsePhase1(
   }
 }
 
+/**
+ * Phase 1b of the Quint parser. Go over each definition of the form
+ * `import ... from '<path>'`, do the following:
+ * 
+ *  - parse the modules that are refferred by each path,
+ *  - add the parsed modules.
+ * 
+ * Cyclic dependencies among different files are reported as errors.
+ */
+export function parsePhase1b(
+  idGen: IdGenerator,
+  sourceResolver: SourceResolver,
+  mainPath: SourceLookupPath,
+  mainPhase1Result: ParserPhase1
+): ParseResult<ParserPhase1> {
+  // we accumulate the source map over all files here
+  let sourceMap = new Map(mainPhase1Result.sourceMap)
+  // The list of modules that have not been been processed yet.
+  // Each element of the list carries the module to be processed and the trail
+  // of sources that led to this module.
+  let worklist: [QuintModule, SourceLookupPath[]][] =
```

Comment:
It doesn't look like you reassign this variable. iiuc, this should be a const. it is a constant reference to a mutable structure: not a mutable reference. afaik, you'll still be able to push into it.

```suggestion
  const worklist: [QuintModule, SourceLookupPath[]][] =
```